pallets/baby-liminal/src/lib.rs

```diff
@@ -197,31 +201,48 @@ pub mod pallet {
         ) -> DispatchResult {
             let who = ensure_signed(origin).map_err(|_| Error::<T>::BadOrigin)?;
             let owner = VerificationKeyOwners::<T>::get(identifier);
-            let previous_key = VerificationKeys::<T>::get(identifier);
 
             if let Some(owner) = owner {
                 ensure!(who == owner, Error::<T>::NotOwner)
             };
 
-            if let Some(previous) = previous_key {
-                ensure!(
-                    previous.len() == key.len(),
-                    Error::<T>::NewKeyMustHaveSameLength
-                )
-            };
-
             ensure!(
                 key.len() <= T::MaximumVerificationKeyLength::get() as usize,
                 Error::<T>::VerificationKeyTooLong
             );
 
             VerificationKeys::<T>::try_mutate_exists(identifier, |value| -> DispatchResult {
                 // should never fail, since length is checked above
-                *value = Some(BoundedVec::try_from(key).unwrap());
+                *value = Some(BoundedVec::try_from(key.clone()).unwrap());
 
                 Ok(())
             })?;
 
+            // TODO : reimburse or lock additional funds
+            let previous_deposit = VerificationKeyDeposits::<T>::get((&who, &identifier))
+                .unwrap_or_else(|| 0u32.into());
```

Comment:
Now, that's sketchy. I understand the idea behind this default is to catch a case where `overwrite_key` was called instead of `insert_key`, so there's no `previous_deposit`, but it feels dangerous. I'd rather prefer is `overwrite_key` was not trying to also cover `store_key` (see my previous comment for alternative).

Reply:
feeling is a weak evidence - if you could give me an example of how this can be dangerous we can change that, but I just don't see it in the situation where the caller locks & unlocks own funds.

Reply:
It's just a bad code that tries to do two different things b/c user might make a mistake. There's no `overwrite` in `HashMap` - `insert` does it if the `key` is already taken.

Reply:
Couldn't we just return in the first `if let` in line `203` if there are no `owner` of `identifier`?

Reply:
Is UPSERT in SQL statements considered bad code then?

Reply:
You're not writing SQL here.

Reply:
Yes, bu surely that does not mean there's nothing to learn from 50+ years of RDBMS history and we have to be reinventing the wheel again either.

Reply:
I think the difference is that when working with dbs you're expecting other clients to update the state, so you can't tell whether the record already exists or not. Here, the design prohibits anyone else but the owner of the entry to modify it so it makes more sense to assume (s)he already knows whether it should be calling `store_*` or `overwrite_*`.